R/utils-code_dependency.R

```diff
@@ -0,0 +1,419 @@
+#' Create Object Dependencies Structure Within Parsed Code
+#'
+#' @description Build up ingredients needed to restore the code required to create a specific object.
+#' @details The relation between objects is assumed to be passed by `<-`, `=` or `->` assignment operators. No other
+#' object creation methods (like `assign`, or `<<-` or any non-standard-evaluation method) are supported. To specify
+#' relations between side-effects and objects, use specific comment tag `# @effect object_name` at the end of the line
+#' in which the side-effect appears. Check examples to see the usage.
+#'
+#' @param parsed_code (`expression`) result of `parse()`
+#' @param object_names (`character(n)`) vector of names of existing objects
+#'
+#' @return A `list` containing 3 elements
+#' - `occurrence` - named `list` by object names with numeric vector as elements indicating calls in which object
+#' appears.
+#' - `cooccurrence` - `list` of the same length as number of calls in `parsed_code`, containing `NULL`s if there is no
+#' co-occurrence between objects, or a `character` vector indicating co-occurrence of objects in specific `parsed_code`
+#' call element. If a character vector, then the first element is the name of the dependent object, and the rest are the
+#' influencing objects
+#' - `effects` - named `list`  by object names with numeric vector as elements indicating which calls has effect on this
+#' object, or NULL if there are no side-effects pointing at this object.
+#'
+#' @examples
+#' \donttest{
```

Comment:
according to some CRAN maintainers we shouldn't have examples of the internal functions. This example is too long anyway IOM

Reply:
yeah, agree, this is here so that I (or reviewer) can test some complicated cases. will remove before the merge! totally

Reply:
33f01bd4291544da9361dd978e34a2efec147ed9